From 01a05a53f40ca2ac5f0af10c6cc0810bee39b792 Mon Sep 17 00:00:00 2001
From: Ross Wang <rosswang@google.com>
Date: Fri, 17 Feb 2017 15:48:08 -0800
Subject: [PATCH] const FieldDescriptorCompare

Clang now validates that <set> comparators must have a const operator():
https://reviews.llvm.org/rL291969

Discussion:
https://groups.google.com/d/msg/protobuf/9W6zFIHaJ-4/9RrfwelpEQAJ
---
 src/google/protobuf/compiler/java/java_file.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/google/protobuf/compiler/java/java_file.cc b/src/google/protobuf/compiler/java/java_file.cc
index cb4503f637..86719f704c 100644
--- a/src/google/protobuf/compiler/java/java_file.cc
+++ b/src/google/protobuf/compiler/java/java_file.cc
@@ -65,7 +65,7 @@ namespace java {
 namespace {
 
 struct FieldDescriptorCompare {
-  bool operator ()(const FieldDescriptor* f1, const FieldDescriptor* f2) {
+  bool operator ()(const FieldDescriptor* f1, const FieldDescriptor* f2) const {
     if(f1 == NULL) {
       return false;
     }
